### PR TITLE
nautilus: os/bluestore: list obj which equals to pend

### PR DESCRIPTION
**Pull request by ifed01**
Just in case - let's backport this to Nautilus...

backport tracker: https://tracker.ceph.com/issues/52771

backport of https://github.com/ceph/ceph/pull/43289
parent tracker: https://tracker.ceph.com/issues/52705

Signed-off-by: Igor Fedotov [igor.fedotov@croit.io](mailto:igor.fedotov@croit.io)

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by aclamk — APPROVED** [review]

**Review by trociny — APPROVED** [review]

File: src/os/bluestore/BlueStore.cc [hunk 1]
```diff
@@ -10112,24 +10120,39 @@ int BlueStore::_collection_list(
...
   ghobject_t coll_range_temp_start, coll_range_temp_end;
   ghobject_t coll_range_start, coll_range_end;
-  bool set_next = false;
   ghobject_t pend;
   bool temp;
 
   if (!pnext)
     pnext = &static_next;
 
+  auto log_latency = make_scope_guard(
```
Comment (neha-ojha) [inline src/os/bluestore/BlueStore.cc:10133]: https://jenkins.ceph.com/job/ceph-pull-requests/90065/consoleFull#-13446221432a811ea2-3e7b-466b-84b4-d13df7e35809

File: src/os/bluestore/BlueStore.cc [hunk 2]
```diff
@@ -30,6 +30,7 @@
 #include "include/stringify.h"
 #include "include/str_map.h"
 #include "include/util.h"
+#include "include/scope_guard.h"
```
Comment (trociny) [inline src/os/bluestore/BlueStore.cc:33]: nit: You seem to add this to a wrong commit. I would expect to see this in "os/bluestore: use scope_guard to log latency". And it is probably worth mentioning in the log message conflicts section.

Comment (ifed01) [reply src/os/bluestore/BlueStore.cc:33]: > nit: You seem to add this to a wrong commit. I would expect to see this in "os/bluestore: use scope_guard to log latency". And it is probably worth mentioning in the log message conflicts section.

yep, fixed. Thanks!

**Review by neha-ojha — APPROVED** [review]

**Comment by ifed01** [comment]
jenkins test make check

**Comment by neha-ojha** [comment]
@yuriw were you able to do a successful nautilus run, the last time you tried?

**Pull request closed without merging**